backend/internal/controllers/router.go

```diff
@@ -23,11 +23,13 @@ func GetRouter(authenticator auth.Authenticator) chi.Router {
 	bookDB := database.BookDatabase{Conn: dbConn}
 	imageDB := database.ImgDatabase{Conn: dbConn}
 	userDB := database.UserDatabase{Conn: dbConn}
+	adminDB := database.AdminDatabase{Conn: dbConn}
 
 	// Set up the controller, which receives and responds to HTTP requests
-	bookController := BookController{Books: bookDB}
+    bookController := BookController{Books: bookDB, Admins: adminDB}
```

Comment:
Ideally you should be able to get rid of this field once the middleware is done.